### DISCUSSION TITLE
Troubleshooting Telegram notifications

### DISCUSSION THREAD

**Discussion by nau-tic** [containrrr/watchtower#816] [Q&A] [answered]
Hi all,

I'm trying to setup Watchtower with notifications via Telegram.

My setup includes these environment variables:
    WATCHTOWER_MONITOR_ONLY: true
    WATCHTOWER_NOTIFICATIONS: shoutrrr
    WATCHTOWER_NOTIFICATION_URL: telegram://xxxxxx@telegram?channels=yyyyy

The container running Watchtower is connected to a user-defined bridge network.

When a notification is triggered, I see following error on the log:
_Failed to send notification via shoutrrr (url=telegram://xxxxxx@telegram?channels=yyyyy):  failed to send notification to "yyyyy", response status code 400 Bad Request_

Any idea what can be wrong? I tried both using "yyyyy" as a channel name as well as my own account.

PS - is there any way to execute bash on the Watchtower container? Would help to troubleshoot connectivity issues.

**Comment by piksel** [comment, marked as answer]
Channel names needs to be prefixed by "@", the other option is chat IDs (so you're not sending the message to the user, you are sending it to the chat that is between the bot and the user), but these are hard to figure out from the the telegram client. You should be able to see them somewhere in the web client if I recall correctly. I have tried to make a bot to simplify this process, but it's not done (and probably won't be as user friendly as I'd hoped, since telegram REALLY didn't have this usage in mind for the bot API).

**Reply by piksel** [reply]
Shoutrrr should probably just add a "@" to any targets that isn't just numbers, as the "@" prefix (even if correctly documented) is just an API detail that the end user shouldn't have to care about....

**Reply by nau-tic** [reply]
Thanks! After many attempts I managed to make it work.

It was an issue with the channel ID. I managed to get the right one corresponding to my account by sending a "/start" to my bot and going directly into https://api.telegram.org/botXXXXXXX/getUpdates (where XXXX is the bot HTTP API token).

One gets the channel id under result -> chat -> id. It's a 10-digit number. Configure it without "@".

I was previously trying with "@my_user_name" or with "yyyyy", being that taken out of the group name http://t.me/joinchat/yyyyy. None of those options worked (also trying with/withouth "@").

On how to get the bot to send notifications to a chat remains open. That will be for another time.

**Reply by piksel** [reply]
Channels can also have a "username", which you can use with a "@" prefix. That name can be set and displayed in the regular telegram client. All chats, including group chats and "personal" chats have a chat ID and can be used (if you can figure out the ID).

The channel "username" is the part after "t.me/" in the screenshot and can be used as @shoutrrr:
![image](https://user-images.githubusercontent.com/807383/107440235-97402680-6b33-11eb-921b-57beebf37feb.png)
